### ISSUE TITLE
Cannot control access to ServerWMC Recordings library

### ISSUE THREAD

**Issue by nebulous999** [jellyfin/jellyfin-plugin-serverwmc#10]
**Steps to reproduce:**

1. Click Menu > Dashboard > "Username" > Access
1. Under Library Access, uncheck "Enable access to all libraries"
1. Click Save
1. Sign in as "Username"
1. Click "ServerWMC Recordings"

**Results:**
The list appears empty, regardless of how many recordings are in the library.
Looking at the Jellyfin log, it contains a line similar to this:
> [2020-03-21 21:30:57.019 -06:00] [WRN] [9] App: "Username" is not permitted to access Library "ServerWMC Recordings".

**Remarks**
This issue was occurring consistently for me after I set up a couple of different users, one of which did not have access to all libraries.
After some digging, I believe this is due to the a condition in the MediaBrowser.Api.UserLibrary.ItemsService.GetQueryResult method:
```cs
private QueryResult<BaseItem> GetQueryResult(GetItems request, DtoOptions dtoOptions, User user)
...
            if (!(item is UserRootFolder) && !user.Policy.EnableAllFolders && !isInEnabledFolder)
            {
                Logger.LogWarning("{UserName} is not permitted to access Library {ItemName}.", user.Name, item.Name);
                return new QueryResult<BaseItem>
                {
                    Items = Array.Empty<BaseItem>(),
                    TotalRecordCount = 0,
                    StartIndex = 0
                };
            }
```
https://github.com/jellyfin/jellyfin/blob/master/MediaBrowser.Api/UserLibrary/ItemsService.cs#L228

I would guess that both UserRootFolder isInEnabledFolder are both likely to be **false** for the ServerWMC Recordings library, so seeing the library must depend on user.Policy.EnableAllFolders being **true** currently.

**Expected results:**
When "Enable access to all libraries" is unchecked, the administrator will be able to check a box for "ServerWMC Recordings", which will provide access to the library for the user.

**Comment by PrplHaz4** [comment]
I'm not sure there's a way to tie user-specific permissions to plugins - what do you think of piggybacking off of the built-in "LiveTV" permissions?  (something like, if you have LiveTV permissions, you get ServerWMC Recordings permissions too?)

Edit: Actually, look at `Users -> Select User -> Access -> Channel Access` this should allow you to grant a specific user access to the ServerWMC Recordings Channel.

**Comment by nebulous999** [comment]
Yes @PrplHaz4, I think that would make sense.  Currently though, when "Channels" is unchecked, unchecking "ServerWMC Recordings" there _does_ make the library it disappear for the user.

Either way, to list the _contents_ of the "ServerWMC Recordings" library, the "Enable access to all libraries" checkbox **must** be checked.  So I guess something has to change if you want to allow ServerWMC access to users who don't get access to **all** libraries.

**Comment by PrplHaz4** [comment]
> 
> 
> Yes @PrplHaz4, I think that would make sense. Currently though, when "Channels" is unchecked, unchecking "ServerWMC Recordings" there _does_ make the library it disappear for the user.
> 
> Either way, to list the _contents_ of the "ServerWMC Recordings" library, the "Enable access to all libraries" checkbox **must** be checked. So I guess something has to change if you want to allow ServerWMC access to users who don't get access to **all** libraries.

Ok, looked into this a little further based on your lead.  The only thing I think you missed is that since ServerWMCRecordings is a Channel (not a folder).

Seems this code was touched here to add trimming by folder:
https://github.com/jellyfin/jellyfin/issues/930
https://github.com/jellyfin/jellyfin/issues/1436

I'm closing this in favor of an issue I opened in the server repo here: https://github.com/jellyfin/jellyfin/issues/2719

**Comment by PrplHaz4** [comment]
Should be fixed in 10.5.4+ by https://github.com/jellyfin/jellyfin/pull/2721

**Comment by nebulous999** [comment]
Excellent, thanks!
